src/pudl/analysis/service_territory.py

```diff
@@ -318,43 +240,47 @@ def get_balancing_authority_geometries(ids,
         generating visualizations.
 
     Args:
-        ids (iterable of ints): The set of balancing_authority_id_eia values for which
-            we are compiling geometries.
-        st_eia861 (pandas.DataFrame): The service_territory_eia861 table.
-        ba_assn_eia861 (pandas.DataFrame): The balancing_authority_assn_eia861 table.
+        ids (iterable of ints): A collection of EIA balancing authority IDs.
+        assn (pandas.DataFrame): Association table, relating ``report_date``,
+        ``state``, and ``utility_id_eia`` to each other, as well as the
+            column indicated by ``assn_col`` -- if it's not ``utility_id_eia``.
+        assn_col (str): Label of the dataframe column in ``assn`` that contains
+            the ID of the entities of interest. Should probably be either
+            ``balancing_authority_id_eia`` or ``utility_id_eia``.
+        st_eia861 (pandas.DataFrame): The EIA 861 Service Territory table.
         census_gdf (geopandas.GeoDataFrame): The US Census DP1 county-level geometries
             as returned by get_census2010_gdf().
+        limit_by_state (bool): Whether to require that the counties associated
+            with the balancing authority are inside a state that has also been
+            seen in association with the balancing authority and the utility
+            whose service territory contians the county.
         dissolve (bool): If False, each record in the compiled territory will correspond
             to a single county, with a county-level geometry, and there will be many
             records enumerating all the counties associated with a given
             balancing_authority_id_eia in each year. If dissolve=True, all of the
             county-level geometries for each utility in each year will be merged
             together ("dissolved") resulting in a single geometry and record for each
             balancing_authority-year.
-        limit_by_state (bool): If True, limit the included counties to only those for
-            which data has actually been reported for each utility in conjunction with
-            each balancing authority and state, in the sales_eia861 or other data
-            tables.
 
     Returns:
         geopandas.GeoDataFrame
 
     """
-    ba_gdf = (
-        get_balancing_authority_counties(
-            ba_ids=ids,
+    return (
+        get_territory_fips(
+            ids=ids,
+            assn=assn,
+            assn_col=assn_col,
             st_eia861=st_eia861,
-            ba_assn_eia861=ba_assn_eia861,
             limit_by_state=limit_by_state,
         )
         .pipe(
             add_geometries,
             census_gdf,
             dissolve=dissolve,
-            dissolve_by=["report_date", "balancing_authority_id_eia"]
+            dissolve_by=["report_date", "utility_id_eia"]
```

Comment:
is this supposed to be `dissolve_by=["report_date", assn_col]`?

Reply:
also, again... this function ended up looking really clean!

Reply:
Whoops, yep. Should have been `assn_col`